daemon/k8s_watcher.go

```diff
@@ -1380,7 +1383,7 @@ func (d *Daemon) addCiliumNetworkPolicyV2(ciliumV2Store cache.Store, cnp *cilium
 					Enforcing:   waitForEPsErr == nil,
 					Revision:    rev,
 					LastUpdated: cilium_v2.NewTimestamp(),
-					Annotations: cnp.Annotations,
+					Annotations: serverRuleCpy.Annotations,
```

Comment:
Leave this unchanged. What you want here is the `cnp`, if you use `serverRuleCpy` you will use the most recent annotations which might not correspond to the revision number set on line 1384.

Reply:
Fixed.